MATLAB wrapper failed to build [Ubuntu]

<!--Please only submit issues/bug reports that come with enough information to reproduce them, ideally a unit test that fails, and possible ideas on what might be wrong. -->

<!-- Even better yet, fix the bug and/or documentation, add a unit test, and create a pull request! -->

<!-- This is a channel to report bugs/issues, not a support channel to help install/use/debug your own code. We'd love to help, but just don't have the bandwidth. Please post questions in the GTSAM Google group (https://groups.google.com/forum/#!forum/gtsam-users) -->

## Description

<!-- A clear description of the bug -->

I am trying to build the GTSAM MATLAB wrapper on Ubuntu 22.04.  I've installed all the required dependencies, but am getting a build error when generating `../wrap/gtsam/gtsam_wrapper.cpp`.

I was following this README: https://github.com/borglab/gtsam/blob/develop/matlab/README.md

## Steps to reproduce

After cloning and installing dependencies (CMake, Boost, TBB), I ran these commands:
```
mkdir build
cd build
cmake .. -DGTSAM_INSTALL_MATLAB_TOOLBOX=ON
make install
```

I got this CMake output:
```
-- GTSAM Version: 4.3a0
-- GTSAM is a shared library due to GTSAM_FORCE_SHARED_LIB
-- GTSAM_POSE3_EXPMAP=ON, enabling GTSAM_ROT3_EXPMAP as well
-- Found Eigen version: 3.4.0
-- checking for thread-local storage - found
-- Could NOT find MKL (missing: MKL_INCLUDE_DIR MKL_LIBRARIES) 
-- Building 3rdparty
-- Could NOT find GeographicLib (missing: GeographicLib_LIBRARY_DIRS GeographicLib_LIBRARIES GeographicLib_INCLUDE_DIRS) 
-- Building base
-- Building basis
-- Building geometry
-- Building inference
-- Building symbolic
-- Building discrete
-- Building hybrid
-- Building linear
-- Building nonlinear
-- Building sam
-- Building sfm
-- Building slam
-- Building navigation
-- GTSAM Version: 4.3.0
-- Install prefix: /usr/local
-- Building GTSAM - as a SHARED library
-- Building base_unstable
-- Building geometry_unstable
-- Building linear_unstable
-- Building discrete_unstable
-- Building dynamics_unstable
-- Building nonlinear_unstable
-- Building slam_unstable
-- Building partition_unstable
-- GTSAM_UNSTABLE Version: 4.3.0
-- Install prefix: /usr/local
-- gtwrap Package config : /usr/local/lib/cmake/gtwrap
-- gtwrap version        : 1.0
-- gtwrap CMake path     : /usr/local/lib/cmake/gtwrap
-- gtwrap library path   : /usr/local/lib/gtwrap
-- gtwrap binary path    : /usr/local/bin/gtwrap
-- gtwrap header path    : /usr/local/include/gtwrap
-- Building wrap module gtsam
-- Installing Matlab Toolbox to /usr/local/gtsam_toolbox
-- Wrote /home/jtyler/Documents/GTSAM/build/GTSAMConfig.cmake
-- Wrote /home/jtyler/Documents/GTSAM/build/GTSAM_UNSTABLEConfig.cmake
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- ===============================================================
-- ================  Configuration Options  ======================
--  CMAKE_CXX_COMPILER_ID type                       : GNU
--  CMAKE_CXX_COMPILER_VERSION                       : 11.4.0
--  CMake version                                    : 3.22.1
--  CMake generator                                  : Unix Makefiles
--  CMake build tool                                 : /usr/bin/gmake
-- Build flags                                               
--  Build Tests                                      : Enabled
--  Build examples with 'make all'                   : Enabled
--  Build timing scripts with 'make all'             : Disabled
--  Build shared GTSAM libraries                     : Enabled
--  Put build type in library name                   : Enabled
--  Build libgtsam_unstable                          : Enabled
--  Build GTSAM unstable Python                      : Enabled
--  Build MATLAB Toolbox for unstable                : Disabled
--  Build for native architecture                    : Disabled
--  Build type                                       : Release
--  C compilation flags                              :  -O3 -DNDEBUG
--  C++ compilation flags                            :  -O3 -DNDEBUG
--  Enable Boost serialization                       : ON
--  GTSAM_COMPILE_FEATURES_PUBLIC                    : cxx_std_17
--  GTSAM_COMPILE_OPTIONS_PUBLIC                     : 
--  GTSAM_COMPILE_DEFINITIONS_PUBLIC                 : 
--  GTSAM_COMPILE_OPTIONS_PUBLIC_RELEASE             : 
--  GTSAM_COMPILE_DEFINITIONS_PUBLIC_RELEASE         : 
--  Use System Eigen                                 : OFF (Using version: 3.4.0)
--  Use System Metis                                 : OFF
--  Using Boost version                              : 1.74.0
--  Use Intel TBB                                    : Yes (Version: 2021.5.0)
--  Eigen will use MKL                               : MKL not found
--  Eigen will use MKL and OpenMP                    : OpenMP found but GTSAM_WITH_EIGEN_MKL is disabled
--  Default allocator                                : TBB
--  Cheirality exceptions enabled                    : YES
--  Build with ccache                                : No
-- Packaging flags
--  CPack Source Generator                           : TGZ
--  CPack Generator                                  : TGZ
-- GTSAM flags                                               
--  Quaternions as default Rot3                      : Disabled
--  Runtime consistency checking                     : Disabled
--  Build with Memory Sanitizer                      : Disabled
--  Rot3 retract is full ExpMap                      : Enabled
--  Pose3 retract is full ExpMap                     : Enabled
--  Allow features deprecated in GTSAM 4.3           : Enabled
--  Metis-based Nested Dissection                    : Enabled
--  Use tangent-space preintegration                 : Enabled
-- MATLAB toolbox flags
--  Install MATLAB toolbox                           : Enabled
--  MATLAB root                                      : /usr/local/MATLAB/R2023a
--  MEX binary                                       : /usr/local/MATLAB/R2023a/bin/glnxa64/mex
-- Python toolbox flags                                      
--  Build Python module with pybind                  : Disabled
-- ===============================================================
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jtyler/Documents/GTSAM/build
```

But, then this build error:
```
[100%] Generating ../wrap/gtsam/gtsam_wrapper.cpp
[MatlabWrapper] Ignoring classes: ['gtsam::Point2', 'gtsam::Point3', 'gtsam::CustomFactor']
[100%] Building CXX object matlab/CMakeFiles/gtsam_matlab_wrapper.dir/__/wrap/gtsam/gtsam_wrapper.cpp.o
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h: In instantiation of ‘void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}; _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>]’:
/usr/include/c++/11/bits/alloc_traits.h:516:17:   required from ‘static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}; _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor> >]’
/usr/include/c++/11/bits/shared_ptr_base.h:519:39:   required from ‘std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}; _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; _Alloc = std::allocator<std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor> >; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr_base.h:650:16:   required from ‘std::__shared_count<_Lp>::__shared_count(_Tp*&, std::_Sp_alloc_shared_tag<_Alloc>, _Args&& ...) [with _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; _Alloc = std::allocator<std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor> >; _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr_base.h:1342:14:   required from ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor> >; _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}; _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/11/bits/shared_ptr.h:409:59:   required from ‘std::shared_ptr<_Tp>::shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor> >; _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}; _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>]’
/usr/include/c++/11/bits/shared_ptr.h:862:14:   required from ‘std::shared_ptr<_Tp> std::allocate_shared(const _Alloc&, _Args&& ...) [with _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; _Alloc = std::allocator<std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor> >; _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}]’
/usr/include/c++/11/bits/shared_ptr.h:878:39:   required from ‘std::shared_ptr<_Tp> std::make_shared(_Args&& ...) [with _Tp = std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>; _Args = {std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >}]’
/home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:14232:109:   required from here
/usr/include/c++/11/ext/new_allocator.h:162:11: error: no matching function for call to ‘std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::pair(std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >)’
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:452:9: note: candidate: ‘template<class ... _Args1, long unsigned int ..._Indexes1, class ... _Args2, long unsigned int ..._Indexes2> std::pair<_T1, _T2>::pair(std::tuple<_Args1 ...>&, std::tuple<_Args2 ...>&, std::_Index_tuple<_Indexes1 ...>, std::_Index_tuple<_Indexes2 ...>) [with _Args1 = {_Args1 ...}; long unsigned int ..._Indexes1 = {_Indexes1 ...}; _Args2 = {_Args2 ...}; long unsigned int ..._Indexes2 = {_Indexes2 ...}; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  452 |         pair(tuple<_Args1...>&, tuple<_Args2...>&,
      |         ^~~~
/usr/include/c++/11/bits/stl_pair.h:452:9: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   ‘std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >’ is not derived from ‘std::tuple<_Tps ...>’
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:387:9: note: candidate: ‘template<class ... _Args1, class ... _Args2> std::pair<_T1, _T2>::pair(std::piecewise_construct_t, std::tuple<_Args1 ...>, std::tuple<_Args2 ...>) [with _Args1 = {_Args1 ...}; _Args2 = {_Args2 ...}; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  387 |         pair(piecewise_construct_t, tuple<_Args1...>, tuple<_Args2...>);
      |         ^~~~
/usr/include/c++/11/bits/stl_pair.h:387:9: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 3 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:381:28: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_MoveConstructiblePair<_U1, _U2>() && (! std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_ImplicitlyMoveConvertiblePair<_U1, _U2>())), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(std::pair<_U1, _U2>&&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_MoveConstructiblePair<_U1, _U2>() && (! std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_ImplicitlyMoveConvertiblePair<_U1, _U2>())), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  381 |         explicit constexpr pair(pair<_U1, _U2>&& __p)
      |                            ^~~~
/usr/include/c++/11/bits/stl_pair.h:381:28: note:   template argument deduction/substitution failed:
/usr/include/c++/11/bits/stl_pair.h:380:38: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  380 |                          bool>::type=false>
      |                                      ^~~~~
/usr/include/c++/11/bits/stl_pair.h:371:19: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_MoveConstructiblePair<_U1, _U2>() && std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_ImplicitlyMoveConvertiblePair<_U1, _U2>()), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(std::pair<_U1, _U2>&&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_MoveConstructiblePair<_U1, _U2>() && std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_ImplicitlyMoveConvertiblePair<_U1, _U2>()), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  371 |         constexpr pair(pair<_U1, _U2>&& __p)
      |                   ^~~~
/usr/include/c++/11/bits/stl_pair.h:371:19: note:   template argument deduction/substitution failed:
/usr/include/c++/11/bits/stl_pair.h:370:38: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  370 |                          bool>::type=true>
      |                                      ^~~~
/usr/include/c++/11/bits/stl_pair.h:361:28: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(_MoveConstructiblePair<_U1, _U2>() && (! _ImplicitlyMoveConvertiblePair<_U1, _U2>())), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(_U1&&, _U2&&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<true, _T1, _T2>::_MoveConstructiblePair<_U1, _U2>() && (! std::_PCC<true, _T1, _T2>::_ImplicitlyMoveConvertiblePair<_U1, _U2>())), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  361 |         explicit constexpr pair(_U1&& __x, _U2&& __y)
      |                            ^~~~
/usr/include/c++/11/bits/stl_pair.h:361:28: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:352:19: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(_MoveConstructiblePair<_U1, _U2>() && _ImplicitlyMoveConvertiblePair<_U1, _U2>()), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(_U1&&, _U2&&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<true, _T1, _T2>::_MoveConstructiblePair<_U1, _U2>() && std::_PCC<true, _T1, _T2>::_ImplicitlyMoveConvertiblePair<_U1, _U2>()), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  352 |         constexpr pair(_U1&& __x, _U2&& __y)
      |                   ^~~~
/usr/include/c++/11/bits/stl_pair.h:352:19: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:343:17: note: candidate: ‘template<class _U2, typename std::enable_if<_CopyMovePair<false, gtsam::DiscreteConditional, _U2>(), bool>::type <anonymous> > std::pair<_T1, _T2>::pair(const _T1&, _U2&&) [with _U2 = _U2; typename std::enable_if<std::_PCC<true, _T1, _T2>::_CopyMovePair<false, _T1, _U2>(), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  343 |        explicit pair(const _T1& __x, _U2&& __y)
      |                 ^~~~
/usr/include/c++/11/bits/stl_pair.h:343:17: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:336:18: note: candidate: ‘template<class _U2, typename std::enable_if<_CopyMovePair<true, gtsam::DiscreteConditional, _U2>(), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(const _T1&, _U2&&) [with _U2 = _U2; typename std::enable_if<std::_PCC<true, _T1, _T2>::_CopyMovePair<true, _T1, _U2>(), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  336 |        constexpr pair(const _T1& __x, _U2&& __y)
      |                  ^~~~
/usr/include/c++/11/bits/stl_pair.h:336:18: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:329:27: note: candidate: ‘template<class _U1, typename std::enable_if<_MoveCopyPair<false, _U1, gtsam::DecisionTreeFactor>(), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(_U1&&, const _T2&) [with _U1 = _U1; typename std::enable_if<std::_PCC<true, _T1, _T2>::_MoveCopyPair<false, _U1, _T2>(), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  329 |        explicit constexpr pair(_U1&& __x, const _T2& __y)
      |                           ^~~~
/usr/include/c++/11/bits/stl_pair.h:329:27: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:322:18: note: candidate: ‘template<class _U1, typename std::enable_if<_MoveCopyPair<true, _U1, gtsam::DecisionTreeFactor>(), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(_U1&&, const _T2&) [with _U1 = _U1; typename std::enable_if<std::_PCC<true, _T1, _T2>::_MoveCopyPair<true, _U1, _T2>(), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  322 |        constexpr pair(_U1&& __x, const _T2& __y)
      |                  ^~~~
/usr/include/c++/11/bits/stl_pair.h:322:18: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:309:28: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_ConstructiblePair<_U1, _U2>() && (! std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_ImplicitlyConvertiblePair<_U1, _U2>())), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(const std::pair<_U1, _U2>&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_ConstructiblePair<_U1, _U2>() && (! std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_ImplicitlyConvertiblePair<_U1, _U2>())), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  309 |         explicit constexpr pair(const pair<_U1, _U2>& __p)
      |                            ^~~~
/usr/include/c++/11/bits/stl_pair.h:309:28: note:   template argument deduction/substitution failed:
/usr/include/c++/11/bits/stl_pair.h:308:38: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  308 |                          bool>::type=false>
      |                                      ^~~~~
/usr/include/c++/11/bits/stl_pair.h:300:19: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_ConstructiblePair<_U1, _U2>() && std::_PCC<((! std::is_same<gtsam::DiscreteConditional, _U1>::value) || (! std::is_same<gtsam::DecisionTreeFactor, _U2>::value)), gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>::_ImplicitlyConvertiblePair<_U1, _U2>()), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(const std::pair<_U1, _U2>&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_ConstructiblePair<_U1, _U2>() && std::_PCC<((! std::is_same<_T1, _U1>::value) || (! std::is_same<_T2, _U2>::value)), _T1, _T2>::_ImplicitlyConvertiblePair<_U1, _U2>()), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  300 |         constexpr pair(const pair<_U1, _U2>& __p)
      |                   ^~~~
/usr/include/c++/11/bits/stl_pair.h:300:19: note:   template argument deduction/substitution failed:
/usr/include/c++/11/bits/stl_pair.h:299:39: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  299 |                           bool>::type=true>
      |                                       ^~~~
/usr/include/c++/11/bits/stl_pair.h:276:26: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(_ConstructiblePair<_U1, _U2>() && (! _ImplicitlyConvertiblePair<_U1, _U2>())), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(const _T1&, const _T2&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<true, _T1, _T2>::_ConstructiblePair<_U1, _U2>() && (! std::_PCC<true, _T1, _T2>::_ImplicitlyConvertiblePair<_U1, _U2>())), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  276 |       explicit constexpr pair(const _T1& __a, const _T2& __b)
      |                          ^~~~
/usr/include/c++/11/bits/stl_pair.h:276:26: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:266:17: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<(_ConstructiblePair<_U1, _U2>() && _ImplicitlyConvertiblePair<_U1, _U2>()), bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair(const _T1&, const _T2&) [with _U1 = _U1; _U2 = _U2; typename std::enable_if<(std::_PCC<true, _T1, _T2>::_ConstructiblePair<_U1, _U2>() && std::_PCC<true, _T1, _T2>::_ImplicitlyConvertiblePair<_U1, _U2>()), bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  266 |       constexpr pair(const _T1& __a, const _T2& __b)
      |                 ^~~~
/usr/include/c++/11/bits/stl_pair.h:266:17: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 2 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:245:26: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<std::__and_<std::is_default_constructible<_U1>, std::is_default_constructible<_Dp>, std::__not_<std::__and_<std::__is_implicitly_default_constructible<_U1>, std::__is_implicitly_default_constructible<_U2> > > >::value, bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair() [with _U1 = _U1; _U2 = _U2; typename std::enable_if<std::__and_<std::is_default_constructible<_U1>, std::is_default_constructible<_U2>, std::__not_<std::__and_<std::__is_implicitly_default_constructible<_U1>, std::__is_implicitly_default_constructible<_U2> > > >::value, bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  245 |       explicit constexpr pair()
      |                          ^~~~
/usr/include/c++/11/bits/stl_pair.h:245:26: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 0 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:232:26: note: candidate: ‘template<class _U1, class _U2, typename std::enable_if<std::__and_<std::__is_implicitly_default_constructible<_U1>, std::__is_implicitly_default_constructible<_U2> >::value, bool>::type <anonymous> > constexpr std::pair<_T1, _T2>::pair() [with _U1 = _U1; _U2 = _U2; typename std::enable_if<std::__and_<std::__is_implicitly_default_constructible<_U1>, std::__is_implicitly_default_constructible<_U2> >::value, bool>::type <anonymous> = <anonymous>; _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  232 |       _GLIBCXX_CONSTEXPR pair()
      |                          ^~~~
/usr/include/c++/11/bits/stl_pair.h:232:26: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/11/bits/c++allocator.h:33,
                 from /usr/include/c++/11/bits/allocator.h:46,
                 from /usr/include/c++/11/string:41,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/ext/new_allocator.h:162:11: note:   candidate expects 0 arguments, 1 provided
  162 |         { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/11/bits/stl_algobase.h:64,
                 from /usr/include/c++/11/bits/char_traits.h:39,
                 from /usr/include/c++/11/string:40,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/types.h:28,
                 from /home/jtyler/Documents/GTSAM/gtsam/global_includes.h:22,
                 from /home/jtyler/Documents/GTSAM/gtsam/base/Vector.h:28,
                 from /home/jtyler/Documents/GTSAM/wrap/matlab.h:25,
                 from /home/jtyler/Documents/GTSAM/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/11/bits/stl_pair.h:315:17: note: candidate: ‘std::pair<_T1, _T2>::pair(std::pair<_T1, _T2>&&) [with _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  315 |       constexpr pair(pair&&) = default;         ///< Move constructor
      |                 ^~~~
/usr/include/c++/11/bits/stl_pair.h:315:22: note:   no known conversion for argument 1 from ‘std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >’ to ‘std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>&&’
  315 |       constexpr pair(pair&&) = default;         ///< Move constructor
      |                      ^~~~~~
/usr/include/c++/11/bits/stl_pair.h:314:17: note: candidate: ‘std::pair<_T1, _T2>::pair(const std::pair<_T1, _T2>&) [with _T1 = gtsam::DiscreteConditional; _T2 = gtsam::DecisionTreeFactor]’
  314 |       constexpr pair(const pair&) = default;    ///< Copy constructor
      |                 ^~~~
/usr/include/c++/11/bits/stl_pair.h:314:22: note:   no known conversion for argument 1 from ‘std::pair<std::shared_ptr<gtsam::DiscreteConditional>, std::shared_ptr<gtsam::DecisionTreeFactor> >’ to ‘const std::pair<gtsam::DiscreteConditional, gtsam::DecisionTreeFactor>&’
  314 |       constexpr pair(const pair&) = default;    ///< Copy constructor
      |                      ^~~~~~~~~~~
make[2]: *** [matlab/CMakeFiles/gtsam_matlab_wrapper.dir/build.make:96: matlab/CMakeFiles/gtsam_matlab_wrapper.dir/__/wrap/gtsam/gtsam_wrapper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:32504: matlab/CMakeFiles/gtsam_matlab_wrapper.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

```


## Environment

<!-- Please provide information about your code environment, things such as OS, language of use (C++, Matlab or Python), version of dependent libraries if using a custom build etc., anything to provide us more information. -->

Ubuntu 22.04.3 LTS

<!-- The more the information, the faster we can help resolve the issue -->

--- a/gtsam/discrete/discrete.i
+++ b/gtsam/discrete/discrete.i
@@ -280,11 +280,11 @@ class DiscreteLookupDAG {
 };
 
 #include <gtsam/discrete/DiscreteFactorGraph.h>
-std::pair<gtsam::DiscreteConditional*, gtsam::DecisionTreeFactor*>
+pair<gtsam::DiscreteConditional*, gtsam::DecisionTreeFactor*>
 EliminateDiscrete(const gtsam::DiscreteFactorGraph& factors,
                   const gtsam::Ordering& frontalKeys);
 
-std::pair<gtsam::DiscreteConditional*, gtsam::DecisionTreeFactor*>
+pair<gtsam::DiscreteConditional*, gtsam::DecisionTreeFactor*>
 EliminateForMPE(const gtsam::DiscreteFactorGraph& factors,
                 const gtsam::Ordering& frontalKeys);
 

